zerver/webhooks/circleci/view.py

```diff
@@ -12,9 +22,39 @@
     "success": "has succeeded",
     "failed": "has failed",
     "canceled": "was canceled",
+    "unauthorized": "was unauthorized",
+    "error": "had an error",
 }
 
-ALL_EVENT_TYPES = list(outcome_to_formatted_status_map.keys())
+GITHUB_COMMIT_LINK = "{target_repository_url}/commit/{commit_sha}"
+
+BITBUCKET_COMMIT_LINK = "{target_repository_url}/commits/{commit_sha}"
+
+GITLAB_COMMIT_LINK = "{web_url}/-/commit/{commit_sha}"
+
+FULL_COMMIT_INFO_TEMPLATE = """
+Triggered on [`{commit_details}`]({commit_link}) on branch `{branch_name}` by {author_name}.
+"""
+
+MANUAL_TRIGGER_INFO_TEMPLATE = """
+Triggered on `{branch_name}`'s HEAD on [{commit_sha}]({commit_link}).
+"""
+
+TAG_TRIGGER_INFO_TEMPLATE = """
+Triggered on the latest tag on [{commit_sha}]({commit_link}).
```

Comment:
Shouldn't `latest` be a variable substitution here? A project can have a variety of ways they use tags.

Reply:
OK, weird.